tests/robot-tests/tests/libs/admin_api.py

```diff
@@ -219,21 +215,41 @@ def delete_test_user(email: str):
 
 
 def _get_user_details_via_api(user_email: str):
-    users = admin_client.get("/api/user-management/users").json()
+    response = admin_client.get("/api/user-management/users")
+
+    if response.status_code == 404:
+        return None
+
+    assert response.status_code == 200, "Error when fetching users via api"
+
+    users = response.json()
 
     matching_users = list(filter(lambda user: user["email"] == user_email, users))
 
-    assert matching_users, f"Could not find user with email {user_email}"
+    if len(matching_users) == 0:
+        return None
+
+    assert len(matching_users) == 1, f"Should only have found one user with email {user_email}"
 
     return matching_users[0]
 
 
 def _get_prerelease_user_details_via_api(user_email: str):
-    users = admin_client.get("/api/user-management/pre-release").json()
+    response = admin_client.get("/api/user-management/pre-release")
+
+    if response.status_code == 404:
```

Comment:
Again with the 404 vs empty list

Reply:
I changed my mind and changed the pre-release users endpoint to return an empty list instead of a 404. So I've also removed the 404 check from the python code.